worldedit-bukkit/src/main/java/com/boydti/fawe/bukkit/util/MinecraftVersion.java

```diff
@@ -0,0 +1,143 @@
+    private final int release;
+
+    /**
+     * Construct a new version with major, minor and release version.
+     *
+     * @param major   Major part of the version, only {@code 1} would make sense.
+     * @param minor   Minor part, full updates, e.g. Nether & Caves & Cliffs
+     * @param release Release, changes for the server software during a minor update.
+     */
+    public MinecraftVersion(int major, int minor, int release) {
+        this.major = major;
+        this.minor = minor;
+        this.release = release;
+    }
+
+    /**
+     * Construct a new version with major and minor version.
+     * The release version is set to 0, therefore ignored.
+     *
+     * @see MinecraftVersion#MinecraftVersion(int, int, int)
+     */
+    public MinecraftVersion(int major, int minor) {
+        this(major, minor, 0);
+    }
+
+    /**
+     * Construct a new version with major, minor and release based on the server version.
+     */
+    public MinecraftVersion() {
+        // Array consists of three version parts, eg. ['v1', '16', 'R3']
+        String[] versionParts = getPackageVersion().split("_");
+
+        if (versionParts.length != 3) {
+            throw new IllegalStateException("Failed to determine minecraft version!");
+        }
+
+        this.major = Integer.parseInt(versionParts[0].substring(1));
+        this.minor = Integer.parseInt(versionParts[1]);
+        this.release = Integer.parseInt(versionParts[2].substring(1));
+    }
+
+    /**
+     * @param other The other version to compare against.
+     * @return {@code true} if this version is equal to the other version.
+     */
+    public boolean isEqual(MinecraftVersion other) {
+        return compareTo(other) == 0;
+    }
+
+    /**
+     * @param other The other version to compare against.
+     * @return {@code true} if this version is higher or equal compared to the other version.
+     */
+    public boolean isEqualOrHigher(MinecraftVersion other) {
+        return compareTo(other) >= 0;
+    }
+
+    /**
+     * @param other The other version to compare against.
+     * @return {@code true} if this version is lower or equal compared to the other version.
+     */
+    public boolean isEqualOrLower(MinecraftVersion other) {
+        return compareTo(other) <= 0;
+    }
+
+    /**
+     * @param other The other version to compare against.
+     * @return {@code true} if this version is higher than the other version.
+     */
+    public boolean isHigher(MinecraftVersion other) {
+        return compareTo(other) > 0;
+    }
+
+    /**
+     * @param other The other version to compare against.
+     * @return {@code true} if this version is lower than to the other version.
+     */
+    public boolean isLower(MinecraftVersion other) {
+        return compareTo(other) < 0;
+    }
+
+    public int getMajor() {
+        return major;
+    }
+
+    public int getMinor() {
+        return minor;
+    }
+
+    public int getRelease() {
+        return release;
+    }
```

Comment:
Do we have a policy on get/set methods? Since Java slowly drops & replaces them by newer features (see Records) I'm a bit opposed to introducing new ones.

Reply:
Currently, basically everything is prefixed, so that would basically break with everything else. I think we can keep it that way.

Reply:
Our policy is to ignore all our policies at all costs then complain when everyone doesn't follow them.